src/system_compositor.cpp

```diff
@@ -99,7 +123,24 @@ void usc::SystemCompositor::run()
                     mir::AbnormalExit("Video driver is blacklisted, exiting"));
             }
 
-            dm_connection = server->the_dm_connection();
+            if (server->get_options()->is_set("from-dm-fd") &&
+                server->get_options()->is_set("to-dm-fd"))
+            {
+                dm_connection = std::make_shared<usc::AsioDMConnection>(
+                    server->get_options()->get("from-dm-fd", -1),
+                    server->get_options()->get("to-dm-fd", -1),
+                    session_switcher);
+            }
+            else if (server->get_options()->is_set("debug-without-dm"))
+            {
+                dm_connection = std::make_shared<NullDMMessageHandler>(
```

Comment:
Looking at this with `gdb` suggests the problem is for a different reason (which makes sense given that the preceding patch would have displayed the segfault if it was about initialization of the `SessionSwitcher` instance):

```gdb
Thread 1 "unity-system-co" received signal SIGSEGV, Segmentation fault.
0x00000000004cee42 in (anonymous namespace)::NullDMMessageHandler::start() ()
(gdb) bt
#0  0x00000000004cee42 in (anonymous namespace)::NullDMMessageHandler::start() ()
#1  0x00000000004cf6be in usc::SystemCompositor::run()::{lambda()#1}::operator()() const ()
#2  0x00000000004cff01 in std::_Function_handler<void (), usc::SystemCompositor::run()::{lambda()#1}>::_M_invoke(std::_Any_data const&) ()
#3  0x00007ffff72ea908 in mir::run_mir(mir::ServerConfiguration&, std::function<void (mir::DisplayServer&)>, std::function<void (int)> const&) ()
   from /usr/lib/x86_64-linux-gnu/libmirserver.so.43
#4  0x00007ffff72f6b9c in mir::Server::run() ()
   from /usr/lib/x86_64-linux-gnu/libmirserver.so.43
#5  0x00000000004cfb62 in usc::SystemCompositor::run() ()
#6  0x00000000004b1bad in main ()
```